File: QuantLibAddin/QuantLibAddin/functions/utilities.cpp

/*
 Copyright (C) 2004 Eric Ehlers

 This file is part of QuantLib, a free-software/open-source library
 for financial quantitative analysts and developers - http://quantlib.org/

 QuantLib is free software: you can redistribute it and/or modify it under the
 terms of the QuantLib license.  You should have received a copy of the
 license along with this program; if not, please email quantlib-dev@lists.sf.net
 The license is also available online at http://quantlib.org/html/license.html

 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the license for more details.
*/

#include <QuantLibAddin/functions/utilities.hpp>

using namespace ObjHandler;

namespace QuantLibAddin {

	std::string QL_LOGFILE(
			const std::string &logFileName) {
		if (setLogFile(logFileName))
			return logFileName;
		else
			return "logging disabled";
	}

	void QL_LOGMESSAGE(
			const std::string &msg) {
		logMessage(msg);
	}

	std::string QL_ANY2STRING(
			const any_ptr &a) {
		return AnyToString(a);
	}

	Properties QL_QUERY(
			const std::string &handle) {
		boost::shared_ptr<Object> object =
				ObjectHandler::instance().retrieveObject(handle);
		if (!object)
				throw Exception("error retrieving object " + handle);
		return object->getProperties();
	}

}

File: QuantLibXL/framework/launcher/Utilities/Utilities.vb

'Copyright (C) 2006 Eric Ehlers

'This file is part of QuantLib, a free-software/open-source library
'for financial quantitative analysts and developers - http://quantlib.org/

'QuantLib is free software: you can redistribute it and/or modify it
'under the terms of the QuantLib license.  You should have received a
'copy of the license along with this program; if not, please email
'<quantlib-dev@lists.sf.net>. The license is also available online at
'<http://quantlib.org/reference/license.html>.

'This program is distributed in the hope that it will be useful, but WITHOUT
'ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
'FOR A PARTICULAR PURPOSE.  See the license for more details.

Module Utilities

    Public Const QUANTLIBXL_DIR As String = "QUANTLIBXL_DIR"
    Public Const EXCEL_PATH As String = "C:\Program Files\Microsoft Office\Office10\EXCEL.EXE"
    Private Const QUANTLIBXL_CONFIG_PATH = "QUANTLIBXL_CONFIG_PATH"

    Function fileExists(ByVal PathName As String) As Boolean
        Try
            Dim fileAttr As FileAttribute
            fileAttr = GetAttr(PathName)
            fileExists = ((fileAttr And FileAttribute.Normal) = FileAttribute.Normal) _
                And ((fileAttr And FileAttribute.Directory) <> FileAttribute.Directory)
            Exit Function
        Catch ex As Exception
        End Try

    End Function

    Function dirExists(ByVal PathName As String) As Boolean
        Try
            Dim fileAttr As FileAttribute
            fileAttr = GetAttr(PathName)
            dirExists = ((fileAttr And FileAttribute.Directory) = FileAttribute.Directory)
            Exit Function
        Catch ex As Exception
        End Try

    End Function

    Function configPath() As String

        configPath = Environ(QUANTLIBXL_CONFIG_PATH)

        If Len(configPath) < 1 Then
            Throw New Exception("Error: environment variable " & _
            QUANTLIBXL_CONFIG_PATH & " is not set.  Please set it " & _
            "equal to the location of the development environment " & _
            "for the QuantLibXL launcher, e.g:" & _
            vbCrLf & vbCrLf & QUANTLIBXL_CONFIG_PATH & _
            "=C:\projects\QuantLibXL\framework\launcher")
        End If

    End Function

End Module
